### DISCUSSION TITLE
C++ exception with description "[System.Error.ECONNREFUSED] Failed to open bus (Connection refused)" thrown

### DISCUSSION THREAD

**Discussion by rocksankalp** [Kistler-Group/sdbus-cpp#216] [Q&A]
Hello All,

i am using libsdbus-c++.so.0.8.3.

i am trying to implement component test for my sdbus service .

my test are passing without  any issue when i run them under docker container with/without  root privilege.

same test app(all test are) failing on Jenkins machine where i am not having root privilege and having root privilege  is not an option for me just for your info. however as per my experience working with docker container user permission shouldn't be the root cause of this problem.

seems connection request  to running dbus-daemon is getting  refused.

i am bringing  up dbus-daemon 1st   before running my test  like below 

sh  'cd build && export DBUS_SYSTEM_BUS_ADDRESS=$(dbus-daemon --print-address --session --fork) && ctest -T test --verbose --no-compress-output'

export DBUS_SYSTEM_BUS_ADDRESS=$(dbus-daemon --print-address --session --fork)# Create a dbus session bus and export as system bus address to simulate a system bus

below are error logs 

**error logs : "C++ exception with description "[System.Error.ECONNREFUSED] Failed to open bus (Connection refused)" thrown in SetUp()."**

in my opinion Dbus-daemon running  but not  responsive/active any more . i am  not sure why is it so? and how can it be kept live & responsive once after brining up.

or to be more precise how can i check to see what makes dbus-daemon process be inactive?

any hints on this would be much appreciated.

log file below.
[connection_refused.txt](https://github.com/Kistler-Group/sdbus-cpp/files/7502387/connection_refused.txt)

**Comment by sangelovic** [comment]
I suggest you minimize this down to a small example (best based on sd-bus API -- I guess only the code for creating a D-Bus connection should be sufficient) and post your issue in the systemd mailing list (and prefix the subject with sd-bus). The exception thrown here is just a wrapper for an error returned by the underlying systemd's sd-bus library.
